tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -34,6 +35,66 @@ using namespace std;
 #pragma clang diagnostic ignored "-Wc++17-extensions" // constexpr if is a C++17 extension
 #endif // __clang__
 
+template <class InIt, class OutIt>
+OutIt last_known_good_adj_diff(InIt first, InIt last, OutIt dest) {
+    if (first == last) {
+        return dest;
+    }
+
+    auto val = *first;
+    *dest    = val;
+
+    for (++first, ++dest; first != last; ++first, ++dest) {
+        auto tmp = *first;
+        *dest    = tmp - val;
+        val      = tmp;
+    }
+
+    return dest;
+}
+
+template <class T>
+void test_case_adj_diff(const vector<T>& input, vector<T>& output_expected, vector<T>& output_actual) {
+    auto actual   = adjacent_difference(input.begin(), input.end(), output_actual.begin());
+    auto expected = last_known_good_adj_diff(input.begin(), input.end(), output_expected.begin());

```

Comment:
These could be `const`.

Reply:
Ok, will fix, but preexisting in some other.

Reply:
Yeah, I hadn't been consistently complaining about this.